File: backend/internal/constants/run_state.go
package constants

import "fmt"

type RunState string

const (
	RunImported       RunState = "imported"
	RunQualityChecked RunState = "quality_checked"
	RunProcessing     RunState = "processing"
	RunProcessed      RunState = "processed"
	RunRejected       RunState = "rejected"
	RunSuperseded     RunState = "superseded"
)

var runTransitions = map[RunState]map[RunState]struct{}{
	RunImported:       {RunQualityChecked: {}, RunRejected: {}},
	RunQualityChecked: {RunProcessing: {}, RunRejected: {}, RunSuperseded: {}},
	RunProcessing:     {RunProcessed: {}, RunRejected: {}},
	RunProcessed:      {},
	RunRejected:       {},
	RunSuperseded:     {},
}

func (s RunState) Valid() bool {
	_, ok := runTransitions[s]
	return ok
}

func (s RunState) CanTransition(target RunState) bool {
	allowed, ok := runTransitions[s]
	if !ok {
		return false
	}
	_, ok = allowed[target]
	return ok
}

func ParseRunState(value string) (RunState, error) {
	state := RunState(value)
	if !state.Valid() {
		return "", fmt.Errorf("unknown run state %q", value)
	}
	return state, nil
}

func RunStates() []RunState {
	return []RunState{RunImported, RunQualityChecked, RunProcessing, RunProcessed, RunRejected, RunSuperseded}
}

File: frontend/src/types/enums/run-state.ts
export type RunState = 'imported' | 'quality_checked' | 'processing' | 'processed' | 'rejected' | 'superseded'
export const RUN_STATE_LABEL: Record<RunState, string> = {
  imported: '已导入', quality_checked: '质量已检', processing: '处理中', processed: '已处理', rejected: '已驳回', superseded: '已替代'
}
export const RUN_TRANSITIONS: Partial<Record<RunState, RunState[]>> = {
  imported: ['quality_checked', 'rejected'], quality_checked: ['processing', 'rejected'], processing: ['processed', 'rejected'], processed: ['superseded']
}
